src/sentry/event_manager.py

```diff
@@ -821,6 +828,23 @@ def _get_or_create_release_many(jobs, projects):
                     pop_tag(job["data"], "dist")
                     set_tag(job["data"], "sentry:dist", job["dist"].name)
 
+            with locks.get(
```

Comment:
This will acquire the lock every time we do this here, yet most of the time we probably can get away without actually locking anything. I believe we can safely change this logic to "read 24h cache key". If that key is `None`, we acquire the lock and do a re-read to see if it's still `None`. if yes, we boost. The 24h key bump can then happen no matter what.